text/0005-white-flag/0005-white-flag.md

```diff
@@ -0,0 +1,103 @@
++ Feature name: `white-flag`
++ Start date: 2020-03-06
++ RFC PR: [iotaledger/protocol-rfcs#0005](https://github.com/iotaledger/protocol-rfcs/pull/5)
+
+# Summary
+
+This RFC is part of a set of protocol changes, [Chrysalis](https://roadmap.iota.org/chrysalis), aiming at improving the
+network before [Coordicide](https://coordicide.iota.org/) is complete.
+
+The feature presented in this RFC, White Flag, allows milestones to confirm conflicting bundles by enforcing
+deterministic ordering of the tangle and applying only the first bundle(s) that does not violate the ledger state.
+
+The content of this RFC is based on [Conflict white flag: Mitigate conflict spamming by ignoring conflicts](https://iota.cafe/t/conflict-white-flag-mitigate-conflict-spamming-by-ignoring-conflicts/233).
+
+# Motivation
+
+<!-- TODO -->
+
+The main motivations:
+
+- Defend against censorship attacks - Conflicts will no longer block bundles from being approved.
+- Make reattachments unnecessary - As long as the network is not saturated, theoretically all bundles should be
+approved. And no bundle will be left behind.
+- Increase TPS - Due to easy node tip selection the network throughput should increase.
+- Increase CTPS - Due to the above, increase in TPS and no left-behinds, we expect CTPS to increase as well.
+
+# Detailed design
+
+<!-- TODO -->
+
+Definitions:
+- confirm
+- ignore
+- reference / indirectly
+
+Let's define a conflicting bundle as a bundle that leads to a negative balance on an address if applied to the current
+ledger state.
+
+In case of conflicting bundles with White Flag, nodes will confirm only one bundle and ignore all the others. For this
+to work, they need to be sure they are all confirming the same bundle; hence, the need for a deterministic ordering
+of the tangle.
+
+## Deterministic ordering of the tangle
+
+<!-- TODO -->
+
+The deterministic ordering of the past-cone of a milestone is a [topological ordering](https://en.wikipedia.org/wiki/Topological_sorting)
+generated by a post-order Depth-First Search by going first through trunks, then branches and finally root.
+The stopping condition of this traversal is reaching bundles that are already confirmed.
```

Comment:
but what is the starting point of the DFS?
i guess the new milestone?
also, it's a bit confusing to say "The stopping condition of this traversal is reaching bundles that are already confirmed."
because all the transactions in this cone are confirmed, it's just that we stop if we reach transactions that were already confirmed by previous milestones

Reply:
Right, already changed that in a future commit, thanks.